connector_amazon/models/amazon_sale_importer.py

```diff
@@ -3,14 +3,204 @@
+            self._get_state_country(sale)
+        if partner:
+            if sale['part_ship'].get('street3') and 'street3' not in \
+                    partner_m._fields:
+                # if street3 doesn't exist in odoo: according to modules
+                sale['part_ship']['street2'] = '%s %s' % (
+                    sale['part_ship']['street2'], sale['part_ship']['street3'])
+            part_ship = partner_m.search([
+                (fieldname, '=', val)
+                for fieldname, val in sale['part_ship'].items()
+                if fieldname in partner_m._fields])
+        else:
+            partner = partner_m.create(sale['partner'])
+        if not part_ship:
+            sale['part_ship']['parent_id'] = partner.id
+            vals = {k: v for k, v in sale['part_ship'].items()
+                    if k in partner_m._fields}
+            part_ship = partner_m.create(vals)
+        return (partner[0], part_ship[0])
+
+    def _complete_products(self, lines, backend):
+        """ - check if product exist in amazon backend
+            - gather shipping price
+            return shipping_price
+        """
+        line_count, shipping_price = 0, 0
+        products_in_exception = []
+        for line in lines:
+            shipping_line = float(line.get('shipping'))
+            if shipping_line:
+                shipping_price += shipping_line
+            binding = self.env['amazon.product'].search(
+                [('external_id', '=', line['sku']),
+                 ('backend_id', '=', backend.id)])
+            if binding:
+                lines[line_count]['product_id'] = binding[0].record_id.id
+            else:
+                products_in_exception.append(line['sku'])
+            line_count += 1
+        if products_in_exception:
+            raise UserError(
+                _("No matching product with these sku '%s' in Amazon binding"
+                  % products_in_exception))
+        return shipping_price
+
+    def _get_state_country(self, sale):
+        """ country is mandatory, not state
+        """
+        country = self.env['res.country'].search(
+            [('code', '=', sale['part_ship'].get('country'))])
+        if not country or not sale['part_ship'].get('country'):
+            raise UserError(
+                _("Unknow country code %s in sale %s " % (
+                    sale['part_ship'].get('code'),
+                    sale['sale']['origin'])))
+        state = False
+        if sale['part_ship'].get('state'):
+            state = self.env['res.country.state'].search(
+                [('code', '=', sale['part_ship'].get('state'))])
+            if not state:
+                raise UserError(
+                    _("Unknown state code %s in sale %s " % (
+                        sale['part_ship'].get('state'),
+                        sale['sale']['origin'])))
+        return(country.id, getattr(state, 'id', state))
```

Comment:
if state is boolean result with `id` attribute is `** AttributeError: 'bool' object has no attribute 'id'` and I need of final value (instead of object) because of this line:
`vals = {k: v for k, v in sale['part_ship'].items() if k in partner_m._fields}`